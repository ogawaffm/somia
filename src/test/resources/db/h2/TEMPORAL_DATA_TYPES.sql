DROP TABLE IF EXISTS TEST.TEMPORAL_DATA_TYPES;
CREATE TABLE TEST.TEMPORAL_DATA_TYPES
(
    DATE_COL             DATE,

    TIME_COL             TIME,
    TIME_0_COL           TIME(0),
    TIME_1_COL           TIME(1),
    TIME_9_COL           TIME(9),

    TIME_WOTZ_COL        TIME WITHOUT TIME ZONE,
    TIME_0_WOTZ_COL      TIME(0) WITHOUT TIME ZONE,
    TIME_1_WOTZ_COL      TIME(1) WITHOUT TIME ZONE,
    TIME_9_WOTZ_COL      TIME(9) WITHOUT TIME ZONE,

    TIME_WTZ_COL         TIME WITH TIME ZONE,
    TIME_0_WTZ_COL       TIME(0) WITH TIME ZONE,
    TIME_1_WTZ_COL       TIME(1) WITH TIME ZONE,
    TIME_9_WTZ_COL       TIME(9) WITH TIME ZONE,

    TIMESTAMP_COL        TIMESTAMP,
    TIMESTAMP_0_COL      TIMESTAMP(0),
    TIMESTAMP_1_COL      TIMESTAMP(1),
    TIMESTAMP_9_COL      TIMESTAMP(9),

    TIMESTAMP_WOTZ_COL   TIMESTAMP WITHOUT TIME ZONE,
    TIMESTAMP_WOTZ_0_COL TIMESTAMP(0) WITHOUT TIME ZONE,
    TIMESTAMP_WOTZ_1_COL TIMESTAMP(1) WITHOUT TIME ZONE,
    TIMESTAMP_WOTZ_9_COL TIMESTAMP(9) WITHOUT TIME ZONE,

    TIMESTAMP_WTZ_COL    TIMESTAMP WITH TIME ZONE,
    TIMESTAMP_0_WTZ_COL  TIMESTAMP(0) WITH TIME ZONE,
    TIMESTAMP_1_WTZ_COL  TIMESTAMP(1) WITH TIME ZONE,
    TIMESTAMP_9_WTZ_COL  TIMESTAMP(9) WITH TIME ZONE,

    UUID_COL             UUID

)