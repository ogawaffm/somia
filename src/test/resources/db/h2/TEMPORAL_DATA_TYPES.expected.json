{
  "name" : "TEMPORAL_DATA_TYPES",
  "catalogName" : "TEST",
  "schemaName" : "TEST",
  "typeName" : "BASE TABLE",
  "comment" : "",
  "columns" : [ {
    "name" : "DATE_COL",
    "type" : {
      "name" : "DATE",
      "typeNumber" : 91,
      "className" : "java.sql.Date",
      "precision" : 10,
      "scale" : 0,
      "isNullable" : true,
      "isAutoIncrement" : false,
      "numericPrecisionRadix" : 0,
      "isCaseSensitive" : false,
      "isSearchable" : true,
      "isSigned" : false,
      "isCurrency" : false,
      "columnDisplaySize" : 10,
      "baseType" : null
    },
    "comment" : "",
    "collation" : "",
    "defaultDefinition" : ""
  }, {
    "name" : "TIME_COL",
    "type" : {
      "name" : "TIME",
      "typeNumber" : 92,
      "className" : "java.sql.Time",
      "precision" : 8,
      "scale" : 0,
      "isNullable" : true,
      "isAutoIncrement" : false,
      "numericPrecisionRadix" : 0,
      "isCaseSensitive" : false,
      "isSearchable" : true,
      "isSigned" : false,
      "isCurrency" : false,
      "columnDisplaySize" : 17,
      "baseType" : null
    },
    "comment" : "",
    "collation" : "",
    "defaultDefinition" : ""
  }, {
    "name" : "TIME_0_COL",
    "type" : {
      "name" : "TIME",
      "typeNumber" : 92,
      "className" : "java.sql.Time",
      "precision" : 8,
      "scale" : 0,
      "isNullable" : true,
      "isAutoIncrement" : false,
      "numericPrecisionRadix" : 0,
      "isCaseSensitive" : false,
      "isSearchable" : true,
      "isSigned" : false,
      "isCurrency" : false,
      "columnDisplaySize" : 17,
      "baseType" : null
    },
    "comment" : "",
    "collation" : "",
    "defaultDefinition" : ""
  }, {
    "name" : "TIME_1_COL",
    "type" : {
      "name" : "TIME",
      "typeNumber" : 92,
      "className" : "java.sql.Time",
      "precision" : 10,
      "scale" : 1,
      "isNullable" : true,
      "isAutoIncrement" : false,
      "numericPrecisionRadix" : 0,
      "isCaseSensitive" : false,
      "isSearchable" : true,
      "isSigned" : false,
      "isCurrency" : false,
      "columnDisplaySize" : 19,
      "baseType" : null
    },
    "comment" : "",
    "collation" : "",
    "defaultDefinition" : ""
  }, {
    "name" : "TIME_9_COL",
    "type" : {
      "name" : "TIME",
      "typeNumber" : 92,
      "className" : "java.sql.Time",
      "precision" : 18,
      "scale" : 9,
      "isNullable" : true,
      "isAutoIncrement" : false,
      "numericPrecisionRadix" : 0,
      "isCaseSensitive" : false,
      "isSearchable" : true,
      "isSigned" : false,
      "isCurrency" : false,
      "columnDisplaySize" : 27,
      "baseType" : null
    },
    "comment" : "",
    "collation" : "",
    "defaultDefinition" : ""
  }, {
    "name" : "TIME_WOTZ_COL",
    "type" : {
      "name" : "TIME",
      "typeNumber" : 92,
      "className" : "java.sql.Time",
      "precision" : 8,
      "scale" : 0,
      "isNullable" : true,
      "isAutoIncrement" : false,
      "numericPrecisionRadix" : 0,
      "isCaseSensitive" : false,
      "isSearchable" : true,
      "isSigned" : false,
      "isCurrency" : false,
      "columnDisplaySize" : 17,
      "baseType" : null
    },
    "comment" : "",
    "collation" : "",
    "defaultDefinition" : ""
  }, {
    "name" : "TIME_0_WOTZ_COL",
    "type" : {
      "name" : "TIME",
      "typeNumber" : 92,
      "className" : "java.sql.Time",
      "precision" : 8,
      "scale" : 0,
      "isNullable" : true,
      "isAutoIncrement" : false,
      "numericPrecisionRadix" : 0,
      "isCaseSensitive" : false,
      "isSearchable" : true,
      "isSigned" : false,
      "isCurrency" : false,
      "columnDisplaySize" : 17,
      "baseType" : null
    },
    "comment" : "",
    "collation" : "",
    "defaultDefinition" : ""
  }, {
    "name" : "TIME_1_WOTZ_COL",
    "type" : {
      "name" : "TIME",
      "typeNumber" : 92,
      "className" : "java.sql.Time",
      "precision" : 10,
      "scale" : 1,
      "isNullable" : true,
      "isAutoIncrement" : false,
      "numericPrecisionRadix" : 0,
      "isCaseSensitive" : false,
      "isSearchable" : true,
      "isSigned" : false,
      "isCurrency" : false,
      "columnDisplaySize" : 19,
      "baseType" : null
    },
    "comment" : "",
    "collation" : "",
    "defaultDefinition" : ""
  }, {
    "name" : "TIME_9_WOTZ_COL",
    "type" : {
      "name" : "TIME",
      "typeNumber" : 92,
      "className" : "java.sql.Time",
      "precision" : 18,
      "scale" : 9,
      "isNullable" : true,
      "isAutoIncrement" : false,
      "numericPrecisionRadix" : 0,
      "isCaseSensitive" : false,
      "isSearchable" : true,
      "isSigned" : false,
      "isCurrency" : false,
      "columnDisplaySize" : 27,
      "baseType" : null
    },
    "comment" : "",
    "collation" : "",
    "defaultDefinition" : ""
  }, {
    "name" : "TIME_WTZ_COL",
    "type" : {
      "name" : "TIME WITH TIME ZONE",
      "typeNumber" : 2013,
      "className" : "java.time.OffsetTime",
      "precision" : 14,
      "scale" : 0,
      "isNullable" : true,
      "isAutoIncrement" : false,
      "numericPrecisionRadix" : 0,
      "isCaseSensitive" : false,
      "isSearchable" : true,
      "isSigned" : false,
      "isCurrency" : false,
      "columnDisplaySize" : 29,
      "baseType" : null
    },
    "comment" : "",
    "collation" : "",
    "defaultDefinition" : ""
  }, {
    "name" : "TIME_0_WTZ_COL",
    "type" : {
      "name" : "TIME WITH TIME ZONE",
      "typeNumber" : 2013,
      "className" : "java.time.OffsetTime",
      "precision" : 14,
      "scale" : 0,
      "isNullable" : true,
      "isAutoIncrement" : false,
      "numericPrecisionRadix" : 0,
      "isCaseSensitive" : false,
      "isSearchable" : true,
      "isSigned" : false,
      "isCurrency" : false,
      "columnDisplaySize" : 29,
      "baseType" : null
    },
    "comment" : "",
    "collation" : "",
    "defaultDefinition" : ""
  }, {
    "name" : "TIME_1_WTZ_COL",
    "type" : {
      "name" : "TIME WITH TIME ZONE",
      "typeNumber" : 2013,
      "className" : "java.time.OffsetTime",
      "precision" : 16,
      "scale" : 1,
      "isNullable" : true,
      "isAutoIncrement" : false,
      "numericPrecisionRadix" : 0,
      "isCaseSensitive" : false,
      "isSearchable" : true,
      "isSigned" : false,
      "isCurrency" : false,
      "columnDisplaySize" : 31,
      "baseType" : null
    },
    "comment" : "",
    "collation" : "",
    "defaultDefinition" : ""
  }, {
    "name" : "TIME_9_WTZ_COL",
    "type" : {
      "name" : "TIME WITH TIME ZONE",
      "typeNumber" : 2013,
      "className" : "java.time.OffsetTime",
      "precision" : 24,
      "scale" : 9,
      "isNullable" : true,
      "isAutoIncrement" : false,
      "numericPrecisionRadix" : 0,
      "isCaseSensitive" : false,
      "isSearchable" : true,
      "isSigned" : false,
      "isCurrency" : false,
      "columnDisplaySize" : 39,
      "baseType" : null
    },
    "comment" : "",
    "collation" : "",
    "defaultDefinition" : ""
  }, {
    "name" : "TIMESTAMP_COL",
    "type" : {
      "name" : "TIMESTAMP",
      "typeNumber" : 93,
      "className" : "java.sql.Timestamp",
      "precision" : 26,
      "scale" : 6,
      "isNullable" : true,
      "isAutoIncrement" : false,
      "numericPrecisionRadix" : 0,
      "isCaseSensitive" : false,
      "isSearchable" : true,
      "isSigned" : false,
      "isCurrency" : false,
      "columnDisplaySize" : 46,
      "baseType" : null
    },
    "comment" : "",
    "collation" : "",
    "defaultDefinition" : ""
  }, {
    "name" : "TIMESTAMP_0_COL",
    "type" : {
      "name" : "TIMESTAMP",
      "typeNumber" : 93,
      "className" : "java.sql.Timestamp",
      "precision" : 19,
      "scale" : 0,
      "isNullable" : true,
      "isAutoIncrement" : false,
      "numericPrecisionRadix" : 0,
      "isCaseSensitive" : false,
      "isSearchable" : true,
      "isSigned" : false,
      "isCurrency" : false,
      "columnDisplaySize" : 39,
      "baseType" : null
    },
    "comment" : "",
    "collation" : "",
    "defaultDefinition" : ""
  }, {
    "name" : "TIMESTAMP_1_COL",
    "type" : {
      "name" : "TIMESTAMP",
      "typeNumber" : 93,
      "className" : "java.sql.Timestamp",
      "precision" : 21,
      "scale" : 1,
      "isNullable" : true,
      "isAutoIncrement" : false,
      "numericPrecisionRadix" : 0,
      "isCaseSensitive" : false,
      "isSearchable" : true,
      "isSigned" : false,
      "isCurrency" : false,
      "columnDisplaySize" : 41,
      "baseType" : null
    },
    "comment" : "",
    "collation" : "",
    "defaultDefinition" : ""
  }, {
    "name" : "TIMESTAMP_9_COL",
    "type" : {
      "name" : "TIMESTAMP",
      "typeNumber" : 93,
      "className" : "java.sql.Timestamp",
      "precision" : 29,
      "scale" : 9,
      "isNullable" : true,
      "isAutoIncrement" : false,
      "numericPrecisionRadix" : 0,
      "isCaseSensitive" : false,
      "isSearchable" : true,
      "isSigned" : false,
      "isCurrency" : false,
      "columnDisplaySize" : 49,
      "baseType" : null
    },
    "comment" : "",
    "collation" : "",
    "defaultDefinition" : ""
  }, {
    "name" : "TIMESTAMP_WOTZ_COL",
    "type" : {
      "name" : "TIMESTAMP",
      "typeNumber" : 93,
      "className" : "java.sql.Timestamp",
      "precision" : 26,
      "scale" : 6,
      "isNullable" : true,
      "isAutoIncrement" : false,
      "numericPrecisionRadix" : 0,
      "isCaseSensitive" : false,
      "isSearchable" : true,
      "isSigned" : false,
      "isCurrency" : false,
      "columnDisplaySize" : 46,
      "baseType" : null
    },
    "comment" : "",
    "collation" : "",
    "defaultDefinition" : ""
  }, {
    "name" : "TIMESTAMP_WOTZ_0_COL",
    "type" : {
      "name" : "TIMESTAMP",
      "typeNumber" : 93,
      "className" : "java.sql.Timestamp",
      "precision" : 19,
      "scale" : 0,
      "isNullable" : true,
      "isAutoIncrement" : false,
      "numericPrecisionRadix" : 0,
      "isCaseSensitive" : false,
      "isSearchable" : true,
      "isSigned" : false,
      "isCurrency" : false,
      "columnDisplaySize" : 39,
      "baseType" : null
    },
    "comment" : "",
    "collation" : "",
    "defaultDefinition" : ""
  }, {
    "name" : "TIMESTAMP_WOTZ_1_COL",
    "type" : {
      "name" : "TIMESTAMP",
      "typeNumber" : 93,
      "className" : "java.sql.Timestamp",
      "precision" : 21,
      "scale" : 1,
      "isNullable" : true,
      "isAutoIncrement" : false,
      "numericPrecisionRadix" : 0,
      "isCaseSensitive" : false,
      "isSearchable" : true,
      "isSigned" : false,
      "isCurrency" : false,
      "columnDisplaySize" : 41,
      "baseType" : null
    },
    "comment" : "",
    "collation" : "",
    "defaultDefinition" : ""
  }, {
    "name" : "TIMESTAMP_WOTZ_9_COL",
    "type" : {
      "name" : "TIMESTAMP",
      "typeNumber" : 93,
      "className" : "java.sql.Timestamp",
      "precision" : 29,
      "scale" : 9,
      "isNullable" : true,
      "isAutoIncrement" : false,
      "numericPrecisionRadix" : 0,
      "isCaseSensitive" : false,
      "isSearchable" : true,
      "isSigned" : false,
      "isCurrency" : false,
      "columnDisplaySize" : 49,
      "baseType" : null
    },
    "comment" : "",
    "collation" : "",
    "defaultDefinition" : ""
  }, {
    "name" : "TIMESTAMP_WTZ_COL",
    "type" : {
      "name" : "TIMESTAMP WITH TIME ZONE",
      "typeNumber" : 2014,
      "className" : "java.time.OffsetDateTime",
      "precision" : 32,
      "scale" : 6,
      "isNullable" : true,
      "isAutoIncrement" : false,
      "numericPrecisionRadix" : 0,
      "isCaseSensitive" : false,
      "isSearchable" : true,
      "isSigned" : false,
      "isCurrency" : false,
      "columnDisplaySize" : 58,
      "baseType" : null
    },
    "comment" : "",
    "collation" : "",
    "defaultDefinition" : ""
  }, {
    "name" : "TIMESTAMP_0_WTZ_COL",
    "type" : {
      "name" : "TIMESTAMP WITH TIME ZONE",
      "typeNumber" : 2014,
      "className" : "java.time.OffsetDateTime",
      "precision" : 25,
      "scale" : 0,
      "isNullable" : true,
      "isAutoIncrement" : false,
      "numericPrecisionRadix" : 0,
      "isCaseSensitive" : false,
      "isSearchable" : true,
      "isSigned" : false,
      "isCurrency" : false,
      "columnDisplaySize" : 51,
      "baseType" : null
    },
    "comment" : "",
    "collation" : "",
    "defaultDefinition" : ""
  }, {
    "name" : "TIMESTAMP_1_WTZ_COL",
    "type" : {
      "name" : "TIMESTAMP WITH TIME ZONE",
      "typeNumber" : 2014,
      "className" : "java.time.OffsetDateTime",
      "precision" : 27,
      "scale" : 1,
      "isNullable" : true,
      "isAutoIncrement" : false,
      "numericPrecisionRadix" : 0,
      "isCaseSensitive" : false,
      "isSearchable" : true,
      "isSigned" : false,
      "isCurrency" : false,
      "columnDisplaySize" : 53,
      "baseType" : null
    },
    "comment" : "",
    "collation" : "",
    "defaultDefinition" : ""
  }, {
    "name" : "TIMESTAMP_9_WTZ_COL",
    "type" : {
      "name" : "TIMESTAMP WITH TIME ZONE",
      "typeNumber" : 2014,
      "className" : "java.time.OffsetDateTime",
      "precision" : 35,
      "scale" : 9,
      "isNullable" : true,
      "isAutoIncrement" : false,
      "numericPrecisionRadix" : 0,
      "isCaseSensitive" : false,
      "isSearchable" : true,
      "isSigned" : false,
      "isCurrency" : false,
      "columnDisplaySize" : 61,
      "baseType" : null
    },
    "comment" : "",
    "collation" : "",
    "defaultDefinition" : ""
  }, {
    "name" : "UUID_COL",
    "type" : {
      "name" : "UUID",
      "typeNumber" : -2,
      "className" : "byte[]",
      "precision" : 16,
      "scale" : 0,
      "isNullable" : true,
      "isAutoIncrement" : false,
      "numericPrecisionRadix" : 0,
      "isCaseSensitive" : false,
      "isSearchable" : true,
      "isSigned" : false,
      "isCurrency" : false,
      "columnDisplaySize" : 32,
      "baseType" : null
    },
    "comment" : "",
    "collation" : "",
    "defaultDefinition" : ""
  } ],
  "indexes" : [ ],
  "collation" : "",
  "primaryKey" : null,
  "bestRowIdentifier" : null
}